src/types/roundware.d.ts

```diff
@@ -0,0 +1,83 @@
+import { Coordinates, IAssetData, IUiConfig, IMixParams, IAudioData } from ".";
+import { IApiClient } from "./api-client";
+import { IAsset, IAssetFilters } from "./asset";
+import { IAudioTrack, IAudioTrackData } from "./audioTrack";
+import { IEnvelope } from "./envelope";
+import { IGeoPosition } from ".";
+import { AudioTrack, IMixer } from "./mixer";
+import { IProject } from ".";
+import { ISession } from ".";
+import { ISpeaker, ISpeakerFilters } from "./speaker";
+import { ITimedAsset } from ".";
+import { ITimedAssetData } from ".";
+import { IUser } from "./user";
+import { ISpeakerData } from "./speaker";
+import { IAssetPool } from "./assetPool";
+
+export interface IRoundware {
+  speakers(): ISpeakerData[];
+  updateLocation(listenerLocation: Coordinates): void;
+  set onUpdateLocation(callback: CallableFunction);
+  set onUpdateAssets(callback: CallableFunction);
+  set onPlayAssets(callback: CallableFunction);
+  _triggerOnPlayAssets(): void;
+  get currentlyPlayingAssets(): unknown;
+  enableGeolocation(mode: number): void;
+  disableGeolocation(): void;
+  connect(): Promise<{ uiConfig: IUiConfig }>;
+  get mixParams(): IMixParams;
+  getAssets(options: object): Promise<unknown[]>;
+  get assetPool(): IAssetPool | undefined;
+  getAssetsFromPool(
+    assetFilter: IAssetFilters,
+    extraParams: object
```

Comment:
These kinds of broad parameters will be really important for us to define more strictly, because they are currently big buckets of a bunch of fields, which get added to and passed around everywhere. So narrowing some `object`, `unknown`, and `CallableFunction` types is an important next step for this work.

Reply:
Great point.
Shreyas, you might come up against this during your work on `roundware-web-template`, so if anything becomes more clear to you as to how best do some of this narrowing, by all means, go for it, but I understand that this requires an in-depth knowledge and experience with the entire codebase in many instances and is understandably out of scope for your initial conversion efforts.